libs/ECS/World/World.hpp

```diff
@@ -0,0 +1,182 @@
+/*
+** EPITECH PROJECT, 2022
+** R-Type
+** File description:
+** World
+*/
+
+#pragma once
+
+#include <concepts>
+#include <cstddef>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <typeindex>
+#include <vector>
+#include "Entity/Entity.hpp"
+#include "Resource/Resource.hpp"
+#include "System/System.hpp"
+#include <unordered_map>
+
+namespace ecs
+{
+    class World final {
+      public:
+        /// @brief The ID type
+        using ID = std::size_t;
+
+        /// @brief This is the map of Entities in the World
+        using EntitiesList = std::map<ID, std::shared_ptr<Entity>>;
```

Comment:
May prefer using EntitiesMap instead of EntitiesList, because you are using a std::map and std::list exist too. This can be confusing.
It's the same comment each time you are talking about list.